pkg/raft/raft.go

```diff
@@ -1033,7 +1038,12 @@ func (r *raft) maybeCommit() bool {
 	// replicas; once an entry from the current term has been committed in this
 	// way, then all prior entries are committed indirectly because of the Log
 	// Matching Property.
-	if !r.raftLog.matchTerm(entryID{term: r.Term, index: index}) {
+	//
+	// Instead of comparing the term of the entry we want to commit with the current term of the leader,
```

Comment:
Try to compress and simplify the comment (e.g. avoid long nested sentences, prefer a linear story). Less is more. For example:

```golang
// The entry at the given index may have already been flushed to storage
// and removed from the in-memory unstable log. To avoid loading the term
// from log storage, use the invariants of idxBeforeLeader. Note that we
// only want to know whether the entry term matches our term.
```